views/tests/test_views.py

```diff
@@ -311,11 +311,14 @@ def test_run_analysis_no_aggregation(self):
             'exposure_layer': buildings_layer.id,
             'hazard_layer': flood_layer.id,
             'keep': False,
-            'extent_option': Analysis.HAZARD_EXPOSURE_CODE
+            'extent_option': Analysis.HAZARD_EXPOSURE_CODE,
+            'language_code': 'en'
         }
 
         form = AnalysisCreationForm(
-            form_data, user=form_data['user'])
+            form_data,
+            user=form_data['user'],
+            language_code=form_data['language_code'])
```

Comment:
CMIIW, I don't think you need this, because it is already present in form_data? (language_code kwargs initializations).

Reply:
Btw, the reason why I put user data in kwargs is because it will not be present in the form data when submitted from the browser (not serializable). it had to be injected from request object